### PR TITLE
feat: fastest path functionality distribution

### LINKED ISSUE #4: [Gateway] Gateway endpoints to API communication
- [x] Pathfinding
- [x] Sensorsim
- [x] SMK API
- [x] Frontend

### PR DESCRIPTION
**Pull request by JackSMunn**
This pull request introduces several important changes to the codebase, focusing on adding new functionalities, improving the testing framework, and configuring code quality tools. The most significant changes include the addition of new API routes, enhancements to the pathfinding service, and the setup of continuous integration workflows for linting and testing.

### New functionalities and improvements:

* [`app/routes/api_routes.py`](diffhunk://#diff-d85da95cf1d2bf708a995b69920e533b5776545164f2c0f791e88bea0e769e9fR1-R18): Added new API routes for testing and calculating the fastest path using the pathfinding service.
* [`app/services/pathfinding_service.py`](diffhunk://#diff-0c9078a6d80f739c2785d0db0e8e2d9ca1cb6da6425195b30e171b737420d0faR1-R61): Implemented the `calculate_fastest_path` function to interact with sensor simulation and pathfinding services.

### Testing enhancements:

* [`app/test/test_forwarder.py`](diffhunk://#diff-db0aef0a3a85d0ff2a6788258d2d3937c9187a48852cb904db304295bfb660beR1-R62): Added tests for the `forward_request` function using fake HTTP clients to simulate success and error scenarios.
* [`app/test/test_gateway.py`](diffhunk://#diff-02566e96156a33593603ed806a3ef67d5d3a2f1fad163347590c0cc160d7c764R1-R121): Added tests for the new API routes, including edge cases and error handling for the pathfinding service.

### Continuous integration setup:

* [`.github/workflows/lint.yml`](diffhunk://#diff-107e910e9f2ebfb9a741fa10b2aa7100cc1fc4f5f3aca2dfe78b905cbd73c0d2R1-R28): Configured a GitHub Actions workflow to run Ruff linter on push and pull request events.
* [`.github/workflows/test.yml`](diffhunk://#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R1-R32): Configured a GitHub Actions workflow to run PyTest on push and pull request events.

### Configuration and documentation updates:

* [`.coveragerc`](diffhunk://#diff-834e9b406d74791ffbafaeec9cc894082cea9739bc347b6ff9f72312c92ebc79R1-R13): Added coverage configuration to omit certain files and show missing lines in the report.
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L72-R72): Updated the development command to correctly reference the `app.main` module.

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,12 @@
 from fastapi import FastAPI
-from app.routes.gateway import router
+from app.routes.api_routes import router  # Updated import
```
Comment (MortenMunk) [inline app/main.py]: Lets remove gippity comment here :laughing: 

```suggestion
from app.routes.api_routes import router
```

**Review by MortenMunk — CHANGES_REQUESTED** [review]
One unnecessary comment, otherwise very good job guys! :denmark: :beer:

File: app/services/pathfinding_service.py
Comment (Jatewo) [inline app/services/pathfinding_service.py:1]: Consider removing gippity comments

File: app/main.py (hunk 1 above)
Comment (JackSMunn) [reply app/main.py]: hehe

File: app/services/pathfinding_service.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+from fastapi import HTTPException
+from app.utils.forwarder import forward_request
+from app.schemas.pathfinding_schema import PathfindingRequest
+
+# Define the URLs for each service.
+SERVICES = {
```
Comment (Jatewo) [inline app/services/pathfinding_service.py]: Can we use env file for these?

File: app/services/pathfinding_service.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    try:
+        path_response = await forward_request(pathfinding_url, "POST", body=payload)
+    except Exception as e:
+        raise HTTPException(
+            status_code=500, 
+            detail="Failed to calculate fastest path from pathfinding service"
+        ) from e
+
+    # Validate the response from pathfinding.
+    if not path_response or not isinstance(path_response, dict):
```
Comment (Jatewo) [inline app/services/pathfinding_service.py]: Use type annotations. I believe it will automatically raise an error but let's test that.

**Review by Jatewo — CHANGES_REQUESTED** [review]
A few things to address. Otherwise, nice work! 👍

**Review by MortenMunk — APPROVED** [review]
lgtm :denmark:

File: app/services/pathfinding_service.py (hunk 3 above)
Comment (JackSMunn) [reply app/services/pathfinding_service.py]: I have done sort of the thing you wanted, using Pydantic instead (setting up schemas)

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,12 @@
 from fastapi import FastAPI
-from app.routes.gateway import router
+from app.routes.api_routes import router
 from app.config import CORS_SETTINGS
 from fastapi.middleware.cors import CORSMiddleware
+import sys
+import os
+import uvicorn
 
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
```
Comment (Jatewo) [inline app/main.py:9]: I believe this is unnecessary if you just use `python -m [path]`

**Review by Jatewo — APPROVED** [review]
# 🦭 - Seal of approval
